practices/dependency-management.md

```diff
@@ -0,0 +1,104 @@
+
+These are defined in the `dependencies` section of the `package.json` file.
+
+Version numbers for run-time dependencies are defined using a tilde `~` plus a full version number in the MAJOR.MINOR.PATCH form. If the PATCH version is 0, it can be omitted.
+
+Specifying the versions in this way ensures that:
+* We don't expose the users of our modules to an unnecessary risk if one of our dependencies releases a minor update that breaks our app.
+* Dependencies in early lifecycle projects (v0.MINOR.PATCH) are traditionally considered of beta quality, or undergoing heavy development, and any minor could potentially contain breaking changes. Limiting the scope to patch updates heavily reduces the chance of our app breaking.
+* The behaviour when installing the dependencies is basically as predictable as possible ([Principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)) without the use of npm-shrinkwrap or similar tools.
+
+Don't use wildcards (i.e. `*`, `x`) in the package numbers or keywords like `latest`. They make the versions less clear, and predicting the outcome of installing the dependencies harder.
+
+
+#### Examples
+
+We do this:
+
+```json
+{
+    "dependencies": {
+        "airbrake": "~1.2.0",
+        "dustmite": "~1.0.0",
+        "shunter": "~4.2.1",
+        "thundermole": "~1.0.3",
+        "truffler": "~1.1"
+    }
+}
+```
+
+We _don't_ do this:
+```json
+{
+    "dependencies": {
+        "airbrake": "latest",
+        "dustmite": "*",
+        "shunter": "~4",
+        "thundermole": "^1.0.3",
+        "truffler": "~1.x"
+    }
+}
+```
+
+[Optional dependencies](https://docs.npmjs.com/files/package.json#optionaldependencies), if present, should also use the same format.
+
+
+### Development dependencies
+
+[Development dependencies](https://docs.npmjs.com/files/package.json#devdependencies) are required during the development, testing, or build process of the app. For example, you may include a test runner, a minifier, a bundler, etc.
+
+These are defined in the `devDependencies` section of the `package.json` file and are not used in production environments.
+
+Version numbers for run-time dependencies are defined using a caret `^` plus a full version number in the MAJOR.MINOR.PATCH form. If the MINOR and PATCH versions are 0, they can be omitted.
+
+
+#### Examples
+
+We do this:
+
+```json
+{
+    "devDependencies": {
+        "jscs": "^2",
+        "mocha": "^3.1",
+        "oxo": "^1.1.1"
+    }
+}
+```
+
+We _don't_ do this:
+```json
+{
+    "dependencies": {
+        "jscs": "latest",
+        "mocha": "*",
+        "toxo": "~1.1.1",
+    }
+}
+```
+
+
+## Dependency management tools
+
+These are some of the tools that we use to help us manage the dependencies of our projects and alert us of any vulnerabilities in them:
+
+* [Node Security Platform](https://nodesecurity.io/)
+* [Snyk](https://snyk.io/)
+* [Bithound](https://www.bithound.io/)
```

Comment:
I know we've _evaluated_ them, but do we actually use Bithound and NSP? Should clarify. May be helpful for people to understand why we've chosen Snyk …